### PR TITLE
support `Modal.confirm({ icon: null })` to hide default Icon

### PR DESCRIPTION
**Pull request by Adamwu1992**
First of all, thank you for your contribution! 😄

New feature please send pull request to feature branch, and rest to master branch.
Pull request will be merged after one of collaborators approve.
Please makes sure that these form are filled before submitting your pull request, thank you!

[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### 👻 What's the background?

希望在调用`Modal.confirm`时可以省略图标

### 💡 Solution

```javascript
Modal.confirm({
  icon: null,
  content: 'hello world'
})
```

### 📝 Changelog description

> Describe changes from user side, and list all potential break changes or other risks.

No break changes.

### ☑️ Self Check before Merge

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by afc163** [comment]
This pull request **introduces 1 alert** when merging dd44b0a6560f9bbe10953b63919be949bfc7d246 into b807859cc325d6cde48eb16e3a258b217c31b04e - [view on LGTM.com](https://lgtm.com/projects/g/ant-design/ant-design/rev/pr-f2d348110956a9a9293fa86a14eceafe82251a70)

**new alerts:**

* 1 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by afc163** [comment]
Mixed other commits, please clean up.

File: components/modal/style/confirm.less [hunk 1]
```diff
@@ -42,6 +42,11 @@
       float: left;
       margin-right: 16px;
       font-size: 22px;
+
+      // `content` after `icon` should set marginLeft
+      + .@{confirm-prefix-cls}-title + .@{confirm-prefix-cls}-content {
```
[the remark this answers is not in the document]
Comment (Adamwu1992) [reply components/modal/style/confirm.less]: Icon 不是有可能不在吗，这句样式是基于`.anticon`的

[the remark this answers is not in the document]
Comment (zombieJ) [reply components/modal/style/confirm.less]: 看漏了，你是对的。

**Comment by yoyo837** [comment]
How about `Modal.confirm({ icon: false })`?

File: components/modal/style/confirm.less [hunk 2]
```diff
@@ -33,7 +33,7 @@
 
     .@{confirm-prefix-cls}-content {
       margin-top: 8px;
-      margin-left: 38px;
+      // margin-left: 38px;
```
Comment (afc163) [inline components/modal/style/confirm.less]: 不要注释，直接删。

**Review by afc163 — APPROVED** [review]

**Pull request merged**